Add Knative Serving architecture page (#5545)
knative/docs: docs/serving/architecture.md

|----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Activator      | The activator is part of the **data-plane**. It is responsible to queue incoming requests (if a `Knative Service` is scaled-to-zero). It communicates with the autoscaler to bring scaled-to-zero Services back up and forward the queued requests. Additional details can be found [here](https://github.com/knative/serving/blob/main/docs/scaling/SYSTEM.md). |
| Autoscaler     | The autoscaler is responsible to scale the `Knative Services` based on configuration, metrics and incoming requests.                                                                                                                                                                                                                                             |

> | Controller     | It is the controller's job to manage the state of `Knative Resources` within the cluster. It watches several objects, manages the live cycle of dependent resources, and updates the resource state.                                                                                                                                                             |

Reviewer: ```suggestion
| Controller     | The controller manages the state of Knative resources within the cluster. It watches several objects, manages the lifecycle of dependent resources, and updates the resource state.                                                                                                                                                             |
```

Reviewer: Might be worth linking to the Kubernetes definition of controllers...

> | Domain-Mapping | The domain-mapping controller is responsible to manage `DomainMapping` resources. It is an extension of the above `Controller`.                                                                                                                                                                                                                                  |

Reviewer: Might mention that this is an alpha component and might or might not be installed.

> | Queue-Proxy    | The Queue-Proxy lives as a `sidecar-container` aside from the `user-container` in a `Knative-Service`. It is responsible to collect metrics and making sure, requests are sent to the `user-container` in the configured rate. It can also act as a queue if necessary, similar to the Activator.                                                                |

Reviewer: ```suggestion
| Queue-Proxy    | The Queue-Proxy lives as a `sidecar-container` aside from the `user-container` in a Knative Service. It is responsible to collect metrics and making sure, requests are sent to the `user-container` in the configured rate. It can also act as a queue if necessary, similar to the Activator.                                                                |
```

Use monospace for specific processes or fields, but not for resource names, and definitely not simply for the term `Knative`.

| Webhooks       | Knative Serving has several webhooks responsible to validate and mutate `Knative Resources`.                                                                                                                                                                                                                                                                     |

## Networking Layer and Ingress

!!! note
    `Ingress` in this case, does not refer to the [Kubernetes Ingress Resource](https://kubernetes.io/docs/concepts/services-networking/ingress/). It refers to the concept of exposing external access to a resource on the cluster. 
    

> Knative Serving depends on a `Networking Layer` that fulfils the [Knative Serving Specification](https://github.com/knative/specs/tree/main/specs/serving). 

Reviewer: Should this link to `networking`?

For this, Knative Serving defines the `KIngress` resource, which acts as an abstraction for different multiple pluggable networking layers. Currently, three networking layers are available and supported:

* [net-kourier](https://github.com/knative-sandbox/net-kourier)
* [net-contour](https://github.com/knative-sandbox/net-contour)
* [net-istio](https://github.com/knative-sandbox/net-istio)

## Traffic flow and DNS

!!! note
    There are fine differences between the different `Networking Layers`, the following section describes the general concept. Also, there are multiple ways to expose your `Ingress Gateway` and configure DNS. Please refer the installation documentation for more information.

![Knative Serving Architecture Ingress](images/serving-architecture-ingress.png)

> * Typically, each `Networking Layer` has a `controller` that is responsible to watch the `KIngress` resources and configure the `Ingress Gateway` accordingly. It will also report back `status` information to the `KIngress` resource.

Author: Maybe thats just the Swiss speaking, but is `information though this resource.` correct?

> * Typically, each `Networking Layer` has a `controller` that is responsible to watch the `KIngress` resources and configure the `Ingress Gateway` accordingly. It will also report back `status` information to the `KIngress` resource.

Reviewer 2: typo - should be `through`

> * Typically, each `Networking Layer` has a `controller` that is responsible to watch the `KIngress` resources and configure the `Ingress Gateway` accordingly. It will also report back `status` information to the `KIngress` resource.

Author: Done.
